s3o_import.py

```diff
@@ -390,21 +398,36 @@ def new_material(tex1, tex2, texsdir, name="Material"):
         #load diffuse texture, plug in UV mapping, link to base color.
         fname = find_in_folder(texsdir, tex1)
         image = bpy.data.images.load(os.path.join(texsdir, fname))
+        image.alpha_mode = 'CHANNEL_PACKED' #spring uses alpha as teamcolor
         tex_node = mat.node_tree.nodes.new('ShaderNodeTexImage')
         tex_node.image = image
         mat.node_tree.links.new(principled.inputs['Base Color'], tex_node.outputs['Color'])
         mat.node_tree.links.new(tex_node.inputs['Vector'], mapping_node.outputs['Vector'])
         
     if tex2 and find_in_folder(texsdir, tex2):
-        #load specular texture, plug in same UV map, set to non colour data and link to specularity.
+        #load reflectivity / emission / data texture, plug in same UV map, set to non colour data and link to appropriate data.
         fname = find_in_folder(texsdir, tex2)
         image = bpy.data.images.load(os.path.join(texsdir, fname))
+        image.alpha_mode = 'STRAIGHT' # The alpha for this file is one bit, but is actual true alpha and applies to both textures once ingame
+        image.colorspace_settings.name = 'Non-Color'
+        image.colorspace_settings.is_data = True
+        
+        # setup texture node associated with new image.
         tex_node = mat.node_tree.nodes.new('ShaderNodeTexImage')
-        tex_node.color_space = 'NONE'
         tex_node.image = image
-        mat.node_tree.links.new(principled.inputs['Specular'], tex_node.outputs['Color'])
+        #tex_node.color_space = 'NONE' #old pre May ~20th, when tex nodes could still have colour spaces associated.
+        
+        #add RGB separation node and hook up associated channels and alpha channel.  R is emission, G is reflectivity (inverse roughness) and B is undefined by default.

```

Comment:
This is very specific of games with PBR (Only Zero-K for the time being).
Also, you have plugged 'Color' to 'Roughness', instead of 'B'

Reply:
That’s what the spring wiki says the format does, so I followed those guidelines.

Blue is not used by default though a game could use it, and I’ve plugged inverted green (invert node has Color in and Color output) to roughness since green at 1 represents full reflectivity, which blender represents with roughness 0

Reply:
Oki!

Reply:
Also you are right, blue is plugged to 'Metal'

Reply:
I'd like it if I could get the nodes to lay themselves out better, would be much easier to see the resulting node tree and less likely to accidentally plug something when inspecting in blender, which I've done at least once.  Pulling them carefully apart though, blue is not attached to anything in my tests in 2.80, and this whole section is not run in 2.79.

I could change this section to use mostly nodes.add_and_link instead of nodes.new, which might (or might not) give a better layout.  Is it worth the effort considering that this currently works, even if it's a bit of a mess when inspected visually?  (If blender just had a built in "rearrange the node tree" function, that would help - but it does not, and an importer should not rely on people having node-wrangler installed.)

Reply:
Nah... It's good like it is